moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -493,13 +494,13 @@ bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void
     ROS_DEBUG_NAMED("collision_detection.fcl", "Actually checking collisions between %s and %s", cd1->getID().c_str(),
                     cd2->getID().c_str());
 
-  double dist_threshold = cdata->req->distance_threshold;
+  thread_local double dist_threshold = cdata->req->distance_threshold;
```

Comment:
similar issues here. It's only ever going to consider the threshold from the first request.